index.go

```diff
@@ -102,11 +104,16 @@ func NewIndex(metricsEndpoint *string, ocpMetaAgent *ocpmetadata.Metadata) *cobr
 				metadata[k] = v
 			}
 			workloads.ConfigSpec.MetricsEndpoints = append(workloads.ConfigSpec.MetricsEndpoints, indexer)
+			if embedded {
```

Comment:
If embedded is not true `ocpConfig` function parameter won't even be used. I was just thinking if we want to provide this embedded feature in `ocp.go` and make it available for other workloads too?

Reply:
I have refactored part of the pr to remove this flag, PTAL again